staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -409,6 +409,37 @@ func (rt *debuggingRoundTripper) CancelRequest(req *http.Request) {
 	}
 }
 
+var knownAuthTypes = map[string]bool{
+	"bearer":    true,
+	"basic":     true,
+	"negotiate": true,
+}
+
+// maskValue masks credential content from authorization headers
+// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization
+func maskValue(key string, value string) string {
```

Comment:
since this is now a separate function, can return early and unnest:

```
if !strings.EqualFold(key, "Authorization") {
  return value
}
```
...
```
if !knownAuthTypes[strings.ToLower(authType)] {
  return "<masked>"
}
```
etc